### ISSUE TITLE
PCB + triplet loss features

### ISSUE THREAD

**Issue by ggalan87** [KaiyangZhou/deep-person-reid#151]
I was wondering why did you use the layers before conv5 as feature map for computing the triplet loss, rather than the later, which are actually used for re-id beyond training.

https://github.com/KaiyangZhou/deep-person-reid/blob/985d9214862aef115a4600b896c682024ea90891/torchreid/models/pcb.py#L226

Is it intentional, or typo error?

Thank's in advance.

EDIT: From what I understand from the original batch-hard triplet loss implementation, loss is computed on their 'final' dimensionality: https://github.com/VisualComputingInstitute/triplet-reid/blob/a538696b89654180f69f19d17679fa48a0346d88/train.py#L287-L295

**Comment by KaiyangZhou** [comment]
The official pcb uses features before conv5 for reid, so it makes sense to impose the constraint on this feature. However, I did not experiment with this design, perhaps using the conv5 features with triplet gets a better performance. Let me know if you have tried the comparison.

**Comment by ggalan87** [comment]
As I can see [here](https://github.com/huanghoujing/beyond-part-models/blob/fdf4fa866ca3e6ded9542026d7174245b899e1e7/bpm/model/PCBModel.py#L54-L68) and [here](https://github.com/huanghoujing/beyond-part-models/blob/fdf4fa866ca3e6ded9542026d7174245b899e1e7/script/experiment/train_pcb.py#L437-L439) PCB uses features after their local conv list (conv per stripe). They also use only CrossEntropy which is the case for part of your loss. Your CrossEntropy uses this part as well.

Otherwise, yes I did experiment with the smaller dimension and the results are up to 1% worse than yours. My configuration included the combined loss (crossentropy+triplet) with default weights.
